core/src/executor/evaluate/mod.rs

```diff
@@ -211,6 +211,14 @@ async fn evaluate_function<'a, T>(
 
     match func {
         // --- text ---
+        Function::Concat(exprs) => {
+            let mut vec: Vec<Evaluated<'_>> = vec![];
+            for expr in exprs {
+                vec.push(eval(expr).await?);
+            }
```

Comment:
In `core` workspace, except for very few cases, `mut` keyword is not allowed.
Let's get familiar with FP styles :)

```rust
let exprs = stream::iter(exprs).then(eval).try_collect().await?;
```

Reply:
This is the first time I've seen stream::iter.. What is the difference between stream::iter(exprs) and expers.iter()?

Reply:
You can think of [Stream](https://docs.rs/futures/latest/futures/prelude/trait.Stream.html) as an asynchronous version of Iterator.

The name is ambiguous, so this time I have an RFC that I want to change to something like AsyncIterator.